### PR TITLE
Re-added the spec_version auto filtering

### LINKED ISSUE #113: Re-implement spec_version auto filtering
Work done by #94 needed to be removed during the branch movement. 

Reimplement this work and remember to modify the tests affected by this.

### PR DESCRIPTION
**Pull request by zrush-mitre**
Closes #113

### CODE REVIEW

**Comment by emmanvg** [comment]
General comments:

- Revert any changes to `initialize_mongodb.py`
- I don't think addressing this issue requires modifying the `default_data.json` nor the new test cases based on the changed data (this helps to keep PRs smaller). Instead I would focus on the minimal changes we can make to the test pass with "spec_version auto filtering".

File: medallion/test/data/default_data.json [hunk 1]
```diff
@@ -158,7 +158,7 @@
                         "name": "Malicious site hosting downloader",
                         "pattern": "[url:value = 'http://z4z10farb.cn/4712']",
                         "pattern_type": "stix",
-                        "spec_version": "2.0",
```
Comment (emmanvg) [inline medallion/test/data/default_data.json:161]: This property can be removed completely. This will better represent a 2.0 object.

**Comment by zrush-mitre** [comment]
I think it's way better to modify tests with the PR. That way, we know it works correctly, and the tests will be more accurate to key edge cases if I do it with the knowledge fresh on my mind.

File: medallion/test/data/initialize_mongodb.py [hunk 2]
```diff
@@ -306,6 +314,20 @@ def reset_db(url="mongodb://root:example@localhost:27017/"):
...
+            "created": datetime_to_float(string_to_datetime("2017-01-27T13:49:53.997Z")),
+            "description": "Poison Ivy",
+            "id": "malware--c0931cc6-c75e-47e5-9036-78fabc95d4ec",
+            "is_family": True,
+            "malware_types": [
+                "remote-access-trojan",
+            ],
+            "modified": datetime_to_float(string_to_datetime("2018-02-23T18:30:00.000Z")),
+            "name": "Poison Ivy",
+            "spec_version": "2.0",
```
Comment (emmanvg) [inline medallion/test/data/initialize_mongodb.py:328]: This property can be removed completely. This will better represent a 2.0 object.

File: medallion/test/data/default_data.json [hunk 3]
```diff
@@ -202,6 +202,19 @@
...
+                        "created": "2017-01-27T13:49:53.997Z",
+                        "description": "Poison Ivy",
+                        "id": "malware--c0931cc6-c75e-47e5-9036-78fabc95d4ec",
+                        "is_family": true,
+                        "malware_types": [
+                            "remote-access-trojan"
+                        ],
+                        "modified": "2018-02-23T18:30:00.000Z",
+                        "name": "Poison Ivy",
+                        "spec_version": "2.0",
```
Comment (emmanvg) [inline medallion/test/data/default_data.json:216]: This property can be removed completely. This will better represent a 2.0 object.

File: medallion/test/test_backends.py [hunk 4]
```diff
@@ -520,7 +550,36 @@ def test_get_object_spec_version(backend):
...
+    )
+
+    assert r.status_code == 200
+    assert r.content_type == MEDIA_TYPE_TAXII_V21
+    objs = r.json
+    assert objs['more'] is False
+    assert len(objs['objects']) == 1
+    for obj in objs['objects']:
+        if obj['id'] == "malware--c0931cc6-c75e-47e5-9036-78fabc95d4ec":
+            obj['modified'] == "2018-02-23T18:30:00.000Z"
```
Comment (emmanvg) [inline medallion/test/test_backends.py:582]: missing `assert` statement.

I would add another assertion like: `assert obj['spev_version'] == "2.1"`

File: medallion/test/test_backends.py [hunk 5]
```diff
@@ -689,7 +748,39 @@ def test_get_manifest_spec_version(backend):
...
+    assert r.status_code == 200
+    assert r.content_type == MEDIA_TYPE_TAXII_V21
+    objs = r.json
+    assert objs['more'] is False
+    assert len(objs['objects']) == 5
+    for obj in objs['objects']:
+        if obj['id'] == "malware--c0931cc6-c75e-47e5-9036-78fabc95d4ec":
+            obj['media_type'] == "application/stix+json;version=2.0"
+        else:
+            obj['media_type'] == "application/stix+json;version=2.1"
```
Comment (emmanvg) [inline medallion/test/test_backends.py:766]: missing `assert` statement

File: medallion/test/test_backends.py [hunk 6]
```diff
@@ -689,7 +748,39 @@ def test_get_manifest_spec_version(backend):
...
```
Comment (emmanvg) [inline medallion/test/test_backends.py:783]: missing `assert` statement
